lib/client-assets.php

```diff
@@ -694,7 +694,11 @@ function gutenberg_default_script_modules() {
 			'fetchpriority' => 'low',
 		);
 
-		gutenberg_register_interactive_script_module_id( $script_module_id );
+		if ( function_exists( 'wp_interactivity' ) && method_exists( wp_interactivity(), 'add_client_navigation_support_to_script_module' ) ) {
+			if ( str_starts_with( $script_module_id, '@wordpress/block-library' ) ) {
+				wp_interactivity()->add_client_navigation_support_to_script_module( $script_module_id );
+			}
+		}
```

Comment:
Nitpick: there's no need to check if `wp_interactivity` exists, since all WordPress versions supported by Gutenberg include it.

```suggestion
		if ( str_starts_with( $script_module_id, '@wordpress/block-library' ) && method_exists( 'WP_Interactivity_API', 'add_client_navigation_support_to_script_module' ) ) {
			wp_interactivity()->add_client_navigation_support_to_script_module( $script_module_id );
		}
```